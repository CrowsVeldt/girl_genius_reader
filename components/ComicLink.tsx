import { useContext } from "react";
import { Text, TouchableOpacity } from "react-native";
import { StyleSheet } from "react-native";
import { DateContext } from "../context/DateContext";
import { formatDate } from "../utils/utilFunctions";

export default function ComicLink({ date, nav }: { date: string; nav: any }) {
  const { changeCurrentDate } = useContext(DateContext);
  return (
    <TouchableOpacity
      style={styles.linkButton}
      onPress={() => {
        changeCurrentDate(date);
        nav.navigate("Home");
      }}
    >
      <Text style={styles.linkText}>{formatDate(date)}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  linkButton: {
    borderColor: "black",
    borderWidth: 1,
  },
  linkText: {
    textAlign: "center",
  },
});
