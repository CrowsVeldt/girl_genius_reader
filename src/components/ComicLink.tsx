import { ContextType, useContext } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import { formatDate } from "../utils/utilFunctions";
import { PageType } from "../utils/types";
import { ComicContext } from "../context/ComicContext";

export default function ComicLink({ page, nav }: { page: PageType; nav: any }) {
  const { changeCurrentPage }: ContextType<typeof ComicContext> =
    useContext(ComicContext);
  return (
    <TouchableOpacity
      style={styles.linkButton}
      onPress={() => {
        changeCurrentPage(page);
        nav.navigate("ComicPage");
      }}
    >
      <Text style={styles.linkNumber}>{page.pageNumber}</Text>
      <Text style={styles.linkText}>{formatDate(page.date)}</Text>
      <Text style={styles.title}>{page.title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  linkButton: {
    width: "100%",
    height: 40,
    flexDirection: "row",
    justifyContent: "space-between",
    borderBottomColor: "black",
    borderBottomWidth: 1,
  },
  linkNumber: {
    marginRight: 6,
    flex: 1,
  },
  linkText: {
    flex: 2,
  },
  title: {
    flex: 3,
    overflow: "visible",
  },
});
