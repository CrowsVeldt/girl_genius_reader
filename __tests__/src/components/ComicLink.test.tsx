import renderer from "react-test-renderer";
import ComicLink from "../../../src/components/ComicLink";
import ComicProvider from "../../../src/context/ComicContext";

describe("<ComicLink />", () => {
  const testPage = {
    date: "20021104",
    title: "",
    pageNumber: 1,
    volumeNumber: 1,
  };

  it("renders the same as last time", () => {
    const tree = renderer.create(
      <ComicProvider>
        <ComicLink page={testPage} nav={jest.mock} />
      </ComicProvider>
    );
    expect(tree).toMatchSnapshot();
  });
});
