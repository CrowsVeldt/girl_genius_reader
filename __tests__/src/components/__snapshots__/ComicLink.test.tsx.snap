// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicLink /> renders the same as last time 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "flex": 1,
        "marginRight": 6,
      }
    }
  >
    Page 1
  </Text>
  <Text
    style={
      {
        "flex": 2,
      }
    }
  >
    2002-11-04
  </Text>
  <Text
    style={
      {
        "flex": 3,
        "overflow": "visible",
      }
    }
  />
</View>
`;
